docs/source/user_guide/data_checks.ipynb

```diff
@@ -261,7 +261,40 @@
     "\n",
     "X = pd.DataFrame(\n",
     "    [[0, 53, 6325, 5], [1, 90, 6325, 10], [2, 90, 18, 20]],\n",
-    "    columns=[\"user_number\", \"cost\", \"revenue\", \"id\"],\n",
+    "    columns=[\"user_number\", \"cost\", \"revenue\", \"revenue_id\"],\n",
+    ")\n",
+    "\n",
+    "id_col_check = IDColumnsDataCheck(id_threshold=0.9)\n",
+    "messages = id_col_check.validate(X)\n",
+    "\n",
+    "errors = [message for message in messages if message[\"level\"] == \"error\"]\n",
+    "warnings = [message for message in messages if message[\"level\"] == \"warning\"]\n",
+    "\n",
+    "for warning in warnings:\n",
+    "    print(\"Warning:\", warning[\"message\"])\n",
+    "\n",
+    "for error in errors:\n",
+    "    print(\"Error:\", error[\"message\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Primary key columns however, can be useful. Though they are ignored from the modeling process, they can be used as an identifier to query on before or after the modeling process. `IDColumnsDataCheck` will also remind you if it finds that the first column of the DataFrame is a primary key. In the given example, 'user_id' is identified as a primary key, while 'revenue_id' was identified as a regular unique identifier. "
```

Comment:
Can you add a bit more clarification for users here about what a primary key column is? There's useful information in here but having a clearer definition will make this more understandable!